IaC/scripts/cleanup_repo_script.sh

```diff
@@ -0,0 +1,136 @@
+                done
+                echo "Sorting list of packages without versions"
+                cat /tmp/packages_without_version_${REPO}_${COMPONENT} | sort | awk '!seen[$0]++' > /tmp/sorted_packages_without_version_${REPO}_${COMPONENT}
+                mv /tmp/sorted_packages_without_version_${REPO}_${COMPONENT} /tmp/packages_without_version_${REPO}_${COMPONENT}
+
+                for EL in el5 el6 el7 el8 src; do
+                    EXCLUDE_LATEST=""
+                    for package in $(cat /tmp/packages_without_version_${REPO}_${COMPONENT}); do
+                        package_path=$(cat /tmp/packages_${REPO}_${COMPONENT} | grep ${EL} | grep "${package}-[0-9]" | sort -r | head -n1)
+                        if ! $(echo $EXCLUDE_LATEST | grep -q "${package}-[0-9]"); then
+                            EXCLUDE_LATEST+=" $(echo $package_path | awk -F "/" '{print $NF}')"
+                        fi
+                    done
+                    echo $EXCLUDE_LATEST >> /tmp/excludes_${REPO}_${COMPONENT}_${EL}
+
+                    EXCLUDE=$(cat /tmp/excludes_${REPO}_${COMPONENT}_${EL})
+                    
+                    if [[ ${DRY_RUN} == "yes" ]]; then
+                        FIND_ARG="echo"
+                        printf "Will be excluded for ${EL} and ${COMPONENT}: $EXCLUDE \n"
+                        printf "Will be removed for ${EL} and ${COMPONENT}: \n"
+                    else
+                        echo "Removing packages except latest versions in ${REPO} in ${COMPONENT} for ${EL}"
+                        FIND_ARG="rm -f"
+                    fi
+                    if [[ ${EL} == 'src' ]]; then
+                        find $REPOPATH/ -type f -name "*.src.rpm" $(printf "! -name %s " $(printf "! -name %s " $EXCLUDE)) -exec $FIND_ARG {} \;
+                    else
+                        find $REPOPATH/ -type f -name "*.${EL}.*.rpm" $(printf "! -name %s " $(printf "! -name %s " $EXCLUDE)) -exec $FIND_ARG {} \;
+                    fi
+                done
+            fi
+
+            if [[ ${DRY_RUN} == "no" ]]; then
+                for RHEL_ARCH in $(find ${REPOPATH}/ -maxdepth 3 -type d \( -name "*x86_64" -o -name "*noarch" -o -name "i386" -o -name "SRPMS" \)); do
+                    if [ -f ${RHEL_ARCH}/repodata/repomd.xml.asc ]; then
+                        rm -f  ${RHEL_ARCH}/repodata/repomd.xml.asc
+                    fi
+                    gpg --detach-sign --armor --passphrase $PASSWORD ${RHEL_ARCH}/repodata/repomd.xml
+                    createrepo --update ${RHEL_ARCH}/
+                done
+            fi
+
+            rm /tmp/excludes_${REPO}_${COMPONENT}_${EL} /tmp/packages_without_version_${REPO}_${COMPONENT}
+        done
+    done
+}
+
+REPO_COMPONENT="all"
+REPO_LIST="tools"
+OLDER_THAN="365"
+MODE="latest"
+DRY_RUN="no"
+
+EXCLUDE_LIST="
+experimental=percona-xtrabackup-test-80-8.0.5-1 percona-backup-mongodb-agent-0.5.0-1
+testing=patchelf-0.12-2
+laboratory= 
+"
```

Comment:
the exclude list isn't full. Please find what packages we shouldn't remove but they were not updated for a long time
Example - sysbnech